### DISCUSSION TITLE
Python "Fernet" encryption , Micropython "cryptolib.aes" decryption

### DISCUSSION THREAD

**Discussion by nastiliano** [micropython/micropython#16362] [General]
Hello,

I'm sending an encrypted TPC socket from my PC in Python via WiFi and I'd like to drecrypt it in my ESP32 using the "cryptolib" library.
I've trying a little bit...

Is it possible?

Thank you very much

Best regards.

**Comment by Josverl** [comment]
AFAIK There is no pure Python implementation of Fernet encryption, or in MicroPython.

For MicroPython, you'd need to implement Fernet encryption manually or adapt existing implementations to work within its constraints. For more details on Fernet encryption, refer to the [cryptography documentation](https://cryptography.io/en/latest/fernet/).

It may be better/simpler to use an encryption protocol that is already supported on both sides.

**Reply by nastiliano** [reply]
Ok.  Thank you very much!
